nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -636,12 +636,24 @@ def _checkpointed_attention_forward(
         """Forward method with activation checkpointing."""
 
         def custom_forward(*inputs):
-            query_layer = inputs[0]
-            key_layer = inputs[1]
-            value_layer = inputs[2]
-            attention_mask = inputs[3]
-            rotary_pos_emb = inputs[4]
-            relative_position_bias = inputs[5]
+            if len(inputs) == 7:
```

Comment:
This looks very brittle.Perhaps we can use here the same solution we used in `nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py` with `self._get_forward_output_only_func` ?

Reply:
I checked the `self._get_forward_output_only_func` method, which has a help function to convert dictionary arguments to positional arguments. Here I have to translate positional arguments into dictionary arguments. Not sure what's a better solution.